contracts/near/eth2-client/src/tests/integration_tests.rs

```diff
@@ -0,0 +1,98 @@
+#[cfg(test)]
+mod integration_tests {
+    use crate::tests::utils::*;
+    use borsh::BorshSerialize;
+    use eth2_utility::types::InitInput;
+    use eth_types::H256;
+    use near_sdk::ONE_NEAR;
+    use near_units::*;
+    use workspaces::operations::Function;
+    use workspaces::prelude::*;
+    use workspaces::{network::Sandbox, Account, Contract, Worker};
+
+    const WASM_FILEPATH: &str = "../res/eth2_client.wasm";
+
+    async fn initialize_client(
+        init_input: InitInput,
+    ) -> anyhow::Result<(Account, Contract, Worker<Sandbox>)> {
+        let worker = workspaces::sandbox().await?;
+        let wasm = std::fs::read(WASM_FILEPATH)?;
+        let contract = worker.dev_deploy(&wasm).await?;
+
+        // create accounts
+        let owner = worker.root_account()?;
+        let alice = owner
+            .create_subaccount(&worker, "alice")
+            .initial_balance(parse_near!("30 N"))
+            .transact()
+            .await?
+            .into_result()?;
+
+        alice
+            .call(&worker, contract.id(), "init")
+            .args(init_input.try_to_vec()?)
+            .transact()
+            .await?;
+        Ok((alice, contract, worker))
+    }
+
+    #[tokio::test]
+    async fn test_gas_usage_of_submit_beacon_chain_light_client_update() -> anyhow::Result<()> {
+        let (headers, updates, init_input) = get_kiln_test_data(Some(InitOptions {
+            validate_updates: false,
+            verify_bls_signatures: false,
+            hashes_gc_threshold: 51000,
+            max_submitted_blocks_by_account: 7000,
+            trusted_signer: None,
+        }));
+        let (alice, contract, worker) = initialize_client(init_input).await?;
+
+        alice
+            .call(&worker, contract.id(), "register_submitter")
+            .deposit(10 * ONE_NEAR)
+            .transact()
+            .await?;
+
+        let num_of_blocks_to_submit = 32;
+        let headers = &headers.as_slice()[1..num_of_blocks_to_submit];
+        for headers_chunk in headers.chunks(50) {
```

Comment:
I can't understand this code. Are we calling 50 methods in one batch transaction? But before that we take a slice of 32 headers.... And why we take a slice from element number 1. I don't get logic.

Reply:
- The first header in the test data set is the finalized one, so we don't need to submit it.
- I've reduced the `num_of_blocks_to_submit` from 320 to 32 to reduce the time of the ci tests.
- Just edit `num_of_blocks_to_submit` to see how much gas is needed for N blocks
- Yes, the batch size is 50